tests/TestCase/View/Helper/BreadcrumbsHelperTest.php

```diff
@@ -0,0 +1,424 @@
+            ->prepend('The root', '/root', ['data-name' => 'some-name'])
+            ->insertBefore('The root', 'The super root');
+
+        $result = $this->breadcrumbs->getCrumbs();
+        $expected = [
+            [
+                'title' => 'The super root',
+                'url' => null,
+                'options' => []
+            ],
+            [
+                'title' => 'The root',
+                'url' => '/root',
+                'options' => ['data-name' => 'some-name']
+            ],
+            [
+                'title' => 'Some text',
+                'url' => [
+                    'controller' => 'Some',
+                    'action' => 'text'
+                ],
+                'options' => []
+            ],
+            [
+                'title' => 'Home',
+                'url' => '/',
+                'options' => [
+                    'class' => 'first'
+                ]
+            ]
+        ];
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * Test adding crumbs after a specific one
+     * @return void
+     */
+    public function testInsertAfter()
+    {
+        $this->breadcrumbs
+            ->add('Home', '/', ['class' => 'first'])
+            ->prepend('Some text', ['controller' => 'Some', 'action' => 'text'])
+            ->prepend('The root', '/root', ['data-name' => 'some-name'])
+            ->insertAfter('The root', 'The less super root');
+
+        $result = $this->breadcrumbs->getCrumbs();
+        $expected = [
+            [
+                'title' => 'The root',
+                'url' => '/root',
+                'options' => ['data-name' => 'some-name']
+            ],
+            [
+                'title' => 'The less super root',
+                'url' => null,
+                'options' => []
+            ],
+            [
+                'title' => 'Some text',
+                'url' => [
+                    'controller' => 'Some',
+                    'action' => 'text'
+                ],
+                'options' => []
+            ],
+            [
+                'title' => 'Home',
+                'url' => '/',
+                'options' => [
+                    'class' => 'first'
+                ]
+            ]
+        ];
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * Tests the render method
+     * @return void
+     */
+    public function testRender()
+    {
+        $this->breadcrumbs
+            ->add('Home', '/', ['class' => 'first', 'innerAttrs' => ['data-foo' => 'bar']])
+            ->add('Some text', ['controller' => 'tests_apps', 'action' => 'some_method'])
+            ->add('Final crumb', null, ['class' => 'final', 'innerAttrs' => ['class' => 'final-link']]);
+
+        $result = $this->breadcrumbs->render(
+            ['data-stuff' => 'foo and bar'],
+            ['separator' => '<i class="fa fa-angle-right"></i>', 'class' => 'separator', 'templateVars' => ['custom' => 'custom']]
```

Comment:
How does `custom` end up in the resulting HTML? I don't see where the templates are changed.